sam/record.go

```diff
@@ -105,9 +105,12 @@ func IsValidRecord(r *Record) bool {
 // Tag returns an Aux tag whose tag ID matches the first two bytes of tag and true.
 // If no tag matches, nil and false are returned.
 func (r *Record) Tag(tag []byte) (v Aux, ok bool) {
-	for i := range r.AuxFields {
-		if bytes.Compare(r.AuxFields[i][:2], tag) == 0 {
-			return r.AuxFields[i], true
+	if len(tag) != 2 {
+		return
+	}
+	for _, aux := range r.AuxFields {
+		if aux[0] == tag[0] && aux[1] == tag[1] {
```

Comment:
Can you replacing this with `if aux.matches(tag) {` adding the following method:

```
func (a Aux) matches(tag []byte) bool { return a[0] == tag[0] && a[1] == tag[1] }
```
probably in `auxtags.go` at around 463.

This will be inlined, and it improves the readability.